### PR TITLE
Metrics

### PR DESCRIPTION
**Pull request by okedeji**
## What is the purpose of the change

added  metrics to offchain node and exposed metrics endpoint on port 2112

### CODE REVIEW

File: lib/metrics_config.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		if err := http.ListenAndServe(port, nil); err != nil {
+			log.Error().Err(err).Msg("Could not start metric server")
+			return
+		}
+
+		log.Info().Msg("Metrics server stopped")
+	}()
+}
+
+func (metrics *Metrics) IncreamentMetricsCounter(counterName string, address string, topic uint64) {
```
Comment (xmariachi) [inline lib/metrics_config.go:59]: typo `Increament` -> `Increment`

File: main.go [hunk 2]
```diff
@@ -52,15 +49,19 @@ func ConvertEntrypointsToInstances(userConfig lib.UserConfig) error {
...
 	if dotErr := godotenv.Load(); dotErr != nil {
-		log.Error().Err(dotErr).Msg("Error loading .env file")
+		log.Info().Msg("Unable to load .env file")
 	}
 
 	// UNIX Time is faster and smaller than most timestamps
 	zerolog.TimeFieldFormat = zerolog.TimeFormatUnix
 	log.Info().Msg("Starting allora offchain node...")
 
+	metrics := lib.NewMetrics(lib.COUNTER_DATA)
```
Comment (xmariachi) [inline main.go:59]: I think metrics should be conditional to configuration, and by default to be off.

**Review by xmariachi — CHANGES_REQUESTED** [review]
Make metrics configurable + typo, lgtm otherwise

File: main.go (hunk 2 above)
Comment (kpeluso) [reply main.go:59]: Our position is "default: on" and users can opt out by blocking the port

**Review by kpeluso — APPROVED** [review]

File: lib/metrics_config.go (hunk 1 above)
Comment (kpeluso) [reply lib/metrics_config.go]: fixed

**Pull request merged**